src/logion/services/locsynchronization.service.ts

```diff
@@ -0,0 +1,37 @@
+import { injectable } from 'inversify';
+import { LocRequestRepository } from '../model/locrequest.model';
+import { ExtrinsicDataExtractor } from "../services/extrinsic.data.extractor";
+
+import { BlockExtrinsics } from './types/responses/Block';
+
+@injectable()
+export class LocSynchronizer {
+
+    constructor(
+        private extrinsicDataExtractor: ExtrinsicDataExtractor,
+        private locRequestRepository: LocRequestRepository,
+    ) {}
+
+    async updateLocRequests(block: BlockExtrinsics): Promise<void> {
+        const timestamp = this.extrinsicDataExtractor.getBlockTimestamp(block);
+        if(timestamp === undefined) {
+            throw Error("Block has no timestamp");
+        }
+        for(let i = 0; i < block.extrinsics.length; ++i) {
+            const extrinsic = block.extrinsics[i];
+            if(extrinsic.method.pallet === "logionLoc" && extrinsic.method.method === "close") {
+                const locId = extrinsic.args['locId'];
+                const loc = await this.locRequestRepository.findById(locId);
+                if(loc === undefined) {
+                    throw Error(`No LOC with ID ${locId}`);
+                }
+                loc.close(timestamp);
+                await this.locRequestRepository.save(loc);
+            }
+        }
+    }
+
+    async reset() {
+        await this.locRequestRepository.deleteAll();
```

Comment:
(Food for though, not a request for change). This one looks a bit extreme to me, given that this table contains much more info than what is on the chain. What do you think of `UPDATE loc_request SET status='REQUESTED' WHERE status in ('OPEN', 'CLOSED')` ?
We should at least log the number of records that are deleted. Note also that this will delete all related files.

Reply:
Indeed, TBH I was not feeling quite right about this. I think I'll just skip this and let the devs delete (or do whatever they want) the requests themselves. What do you think?

Reply:
> Indeed, TBH I was not feeling quite right about this. I think I'll just skip this and let the devs delete (or do whatever they want) the requests themselves. What do you think?

Agreed ! Let such a process (that is only supposed to sync statuses and timestamps) delete an entire table looks really dangerous to me.

Reply:
I'll remove this in a future PR.